### ISSUE TITLE
The type 'TRequest' must be convertible to 'MediatR.IRequest<TResponse>' in order to use it as parameter 'TRequest' in the generic interface 'MediatR.IPipelineBehavior<in TRequest,TResponse>'

### ISSUE THREAD

**Issue by TarasKovalenko** [DavidRogersDev/ServerFramework#2]
The issue with the latest MediatR version:

![image](https://user-images.githubusercontent.com/2806803/176375139-7d7d2206-6bc5-49aa-833f-1f5ac8914eed.png)

Any idea how we can solve it?
Thanks!

**Comment by DavidRogersDev** [comment]
Ah, a breaking change. To fix that, just change the second Generic constraint to:
`where TRequest : IRequest<TResponse>, IValidateable`

In the other pipeline behaviors, you would add a new generic constraint:
`where TRequest : IRequest<TResponse>`

Fixed in commit [22f4276](https://github.com/DavidRogersDev/ServerFramework/commit/22f4276a3449502818ac550da3f20cc66e7eb253)

Thanks for pointing that out!
I'll close the issue, assuming this solves it for you. Feel free to re-open if it doesn't.
Cheers

**Comment by TarasKovalenko** [comment]
Thank you @DavidRogersDev, now it works well. 
Any idea how we can change `BusinessValidationPipeline` to don't use `Activator.CreateInstance` but use a `new TResponse`?
Because `Activator.CreateInstance` has a significant performance difference versus `new` operator.

**Comment by DavidRogersDev** [comment]
Well. You did make me realize that if the `TResponse` is a `ValidateableResponse` (of the non-generic variety), then I can just "new" one up. Thanks. Fixed in [3f1ee51](https://github.com/DavidRogersDev/ServerFramework/commit/3f1ee5134118ebe717c97b36b8793aafc723dac2)

However, I don't know of any way of creating a generic version of the ValidateableResponse without using Activator.CreateInstance. I've done it before using the Il emitter, but I've found Activator.CreateInstance to be actually faster than that. They must have fine-tuned it a lot, as it is not that slow.

These are some benchmarks which I have done:
create using "new" - negligible (too fast to measure)
Activator.CreateInstance - 560ns
Exception thrown - 5,000ns

So `Activator.CreateInstance` is 10x faster than throwing/catching an exception and only 560ns (that's nano). Unless you have some extreme performance requirements, it should suffice.
But if you do ever figure out a faster way of creating a ValidateableResponse (the generic version), I would be very interested in that.

Cheers
